src/CMakeLists.txt

```diff
@@ -14,15 +14,73 @@ target_include_directories(micm
     $<INSTALL_INTERFACE:include>
 )
 
+set(CMAKE_CUDA_ARCHITECTURES 80)
+
 if(ENABLE_JSON)
   target_link_libraries(micm INTERFACE nlohmann_json::nlohmann_json)
   target_compile_definitions(micm INTERFACE USE_JSON)
 endif()
 
 if(ENABLE_OPENMP)
-  target_link_libraries(micm PRIVATE OpenMP::OpenMP_CXX)
+  target_link_libraries(micm INTERFACE OpenMP::OpenMP_CXX)
 endif()
 
 if(ENABLE_MPI)
-  target_link_libraries(micm PRIVATE MPI::MPI_CXX)
+  target_link_libraries(micm INTERFACE MPI::MPI_CXX)
+endif()
+
+if(ENABLE_OPENACC)
+  add_library(micm_openacc)
+  add_library(musica::micm_openacc ALIAS micm_openacc)
+
+  if(NOT GPU_TYPE)
+    message(FATAL_ERROR "GPU_TYPE is not set or is empty. Please provide a GPU type.")
+  endif()
+
+  set(GPU_ARCH "")
+  if (GPU_TYPE STREQUAL a100)
+    set(GPU_ARCH "cc80")
+  endif()
+  if (GPU_TYPE STREQUAL v100)
+    set(GPU_ARCH "cc70")
+  endif()
+
+  if(NOT GPU_ARCH)
+    message(FATAL_ERROR "GPU_TYPE (${GPU_TYPE}) is not recognized. Available options are [a100, v100].")
+  endif()
+  
+  set(GPU_FLAGS -gpu=${GPU_ARCH},lineinfo -Minfo=accel)
+  message(STATUS "GPU Flags: ${OpenACC_CXX_FLAGS} ${GPU_FLAGS}")
+
+  target_compile_options(micm_openacc PRIVATE ${OpenACC_CXX_FLAGS} ${GPU_FLAGS})
+
+  target_link_libraries(micm_openacc 
+    PRIVATE micm
+  )
+
+  target_link_libraries(micm_openacc 
+    PUBLIC OpenACC::OpenACC_CXX
+  )
 endif()
+
+if(ENABLE_CUDA)
+  add_library(micm_cuda)
+  add_library(musica::micm_cuda ALIAS micm_cuda)
+  
+  target_link_libraries(micm_cuda 
+    PRIVATE micm
+  )
+
+  if(ENABLE_CUDA)
+    target_link_libraries(micm_cuda 
```

Comment:
Shall we add a CMake variable to specify some customized flags for CUDA compilation? What is the default flags used currently?

Reply:
Whatever flags the `cudart` target specifies is what we add. I am fine with adding an option to control flags passed to the GPU, but perhaps we do that in a future PR?

Reply:
Agreed. We can always customize it in the future. Just curious about the default setting here.